src/laser_geometry.cpp

```diff
@@ -437,7 +437,8 @@ const boost::numeric::ublas::matrix<double>& LaserProjection::getUnitVectors_(do
     {
       //check to see if we want to keep the point
       const float range = scan_in.ranges[i];
-      if ((preservative && std::isfinite(ranges(i,0))) || (range < range_cutoff && range >= scan_in.range_min))
```

Comment:
According to [REP 117](http://www.ros.org/reps/rep-0117.html) it can:

> Detections that are too close to the sensor to quantify shall be represented by -Inf. 

and this is the only situation that ```isfinite```  would be checking for because in ```preservative``` mode the +Inf would have been removed in the previous check.